src/zenml/models/v2/base/base.py

```diff
@@ -255,15 +256,35 @@ def get_body(self) -> AnyBody:
 
         Returns:
             The body field of the response.
+
+        Raises:
+            IllegalOperationError: If the user lacks permission to access the
+                entity represented by this response.
         """
+        if not self.body:
+            raise IllegalOperationError(
+                f"Missing permissions to access {type(self).__name__} with "
+                f"ID {self.id}."
+            )
+
         return self.body
 
     def get_metadata(self) -> "AnyMetadata":
         """Fetch the metadata of the entity.
 
         Returns:
             The metadata field of the response.
+
+        Raises:
+            IllegalOperationError: If the user lacks permission to access this
+                entity represented by this response.
         """
+        if not self.body:
+            raise IllegalOperationError(
+                f"Missing permissions to access {type(self).__name__} with "
+                f"ID {self.id}."
+            )
```

Comment:
Same here: raise a different exception if `permission_denied` is set to False ?

Reply:
Done